src/utilmoneystr.cpp

```diff
@@ -71,7 +71,10 @@ bool ParseMoney(const char* pszIn, CAmount& nRet)
         return false;
     if (nUnits < 0 || nUnits > COIN)
         return false;
-    int64_t nWhole = atoi64(strWhole);
+    int64_t nWhole;
+    if (!ParseInt64(strWhole, &nWhole)) {
+        return false;
+    }
```

Comment:
This makes me slightly nervous, because behavior changes if the string begins with a non-digit (`atoi64` returns zero which causes this function to return `true` below (success); `ParseInt64` returns `false` which causes this function to fail and return `false`), but all tests pass, so... But if we want to be safer, we could set `nWhole` to zero (mimicing `atoi`) here instead of returning `false`.